pkg/domain/infra/abi/images.go

```diff
@@ -806,3 +817,32 @@ func localPathFromURI(url *url.URL) (string, error) {
 	}
 	return url.Path, nil
 }
+
+func createSignature(m []byte, dockerReference string, mech signature.SigningMechanism, keyIdentity string) ([]byte, error) {
+	newSig, err := signature.SignDockerManifest(m, dockerReference, mech, keyIdentity)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating new signature")
+	}
+	return newSig, nil
+}
+
+func putSignature(sigStoreDir string, signature []byte, instanceDigest digest.Digest) error {
```

Comment:
(More of a reminder of an option than a review comment: `createSignature`+`putSignature` can be a single function instead of two. Entirely up to you, this works just fine.)

Reply:
Oops, this is not possible if the error reporting regimes are different.